src/MizarWidgets/SamplingWithFrameTrackWidgetTest.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <qnamespace.h>
+
+#include <QApplication>
+#include <QCheckBox>
+#include <QLabel>
+#include <memory>
+
+#include "MizarWidgets/SamplingWithFrameTrackWidget.h"
+
+using ::testing::NotNull;
+
+namespace orbit_mizar_widgets {
+
+constexpr const char* kOrgName = "The Orbit Authors";
+
+constexpr std::string_view kMultiplicityCorrectionEnabledLabel =
+    "Probability of false-alarm for at least one function:";
+constexpr std::string_view kMultiplicityCorrectionDisabledLabel =
+    "Probability of false-alarm for an individual function:";
+
+class SamplingWithFrameTrackWidgetTest : public ::testing::Test {
+ public:
+  SamplingWithFrameTrackWidgetTest() : widget_(std::make_unique<SamplingWithFrameTrackWidget>()) {
+    QApplication::setOrganizationName(kOrgName);
+    QApplication::setApplicationName("SamplingWithFrameTrackInputWidgetTest.InitIsCorrect");
+
+    multiplicity_correction_ = widget_->findChild<QCheckBox*>("multiplicity_correction_");
+    EXPECT_THAT(multiplicity_correction_, NotNull());
+
+    significance_level_label_ = widget_->findChild<QLabel*>("significance_level_label_");
+    EXPECT_THAT(significance_level_label_, NotNull());
+  }
+
+  void ClickMultiplicityCorrectionCheckBox() const {
+    multiplicity_correction_->setChecked(!multiplicity_correction_->isChecked());
+  }
+
+  void ExpectMultiplicityCorrectionEnabledIsCorrectlyShown() const {
+    EXPECT_TRUE(multiplicity_correction_->isChecked());
+    EXPECT_EQ(significance_level_label_->text().toStdString(), kMultiplicityCorrectionEnabledLabel);
+  }
+
+  void ExpectMultiplicityCorrectionDisabledIsCorrectlyShown() const {
+    EXPECT_FALSE(multiplicity_correction_->isChecked());
+    EXPECT_EQ(significance_level_label_->text().toStdString(),
+              kMultiplicityCorrectionDisabledLabel);
+  }
+
+  QCheckBox* multiplicity_correction_;
+  QLabel* significance_level_label_;
+  std::unique_ptr<SamplingWithFrameTrackWidget> widget_;
+};
+
+TEST_F(SamplingWithFrameTrackWidgetTest, MultiplicityCorrectionCheckBoxIsCorrect) {
```

Comment:
I would consider adding another test for the combo box. Sadly I dont think there is a QTest::mouseClick equivalent for combobox, but you can use `setCurrentIndex` and then you would need a public getter for `significance_level_`. 

I dont know if this is worth it, or if it will also be tested by a future PR. (In case you want to get the value of `significance_level_` out of the widget at some point, you will need a getter or a signal anyways)

Reply:
I plan to test this as a part of implementation of the future logic of update button.